pkg/agent/flexvolume/crd/types.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some of the code below came from https://github.com/coreos/etcd-operator
+which also has the apache 2.0 license.
+*/
+
+package crd
+
+import (
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// schemeGroupVersion is group version used to register these objects
+var schemeGroupVersion = schema.GroupVersion{Group: k8sutil.CustomResourceGroup, Version: k8sutil.V1Alpha1}
+
+type Volumeattachment struct {
```

Comment:
Did you look into naming it `VolumeAttachment` to get the casing correct?

Reply:
yes but that doesnt work for TPR. There is an open issue to fix this (https://github.com/rook/rook/issues/1010). So i want this to be solved with that issue. That issue will address the naming of not just Volumeattachment but also Objectstore and Filesystem.

Reply:
#1010 was intended only to be for ObjectStore since we had the discussion after it was merged. I don't think it makes sense to merge this, then immediately rename it in the next PR. 

It's not a common fix in the operator kit, right? We need specific code for each type to handle the TPR case so why not do it with this PR?

Reply:
i want to do this in a more uniform way since it will also apply to other CRDs like filesystem and objectstore. So perhaps we should tailored #1010 to handle other CRDs besides objectstore. I prefer not to handle these in the flexvolume code. I think it is more suitable to be handled somewhere outside, possible kit package. So that when we drop support for 1.6, we can just simply remove that code which only exists in one place.

Reply:
remember that it is not about renaming it. We need to check whether this is a k8s 1.6 or or 1.7 and act accordingly. That is why i think this should be done in a more central way.

Reply:
i think it will need specific code for each crd, but it's find to address after the merge

Reply:
yeah. i rather do that later.. Because i want to isolate and centralize the logic of renaming .. possible be done in the `kit` package.